File: backend/src/entities/wish.php
<?php

class Wish implements JsonSerializable
{
    private $id;
    private $authorId;
    private $content;
    private $done;
    private $timestamp;

    public function __construct($id, $authorId, $content, $done, $timestamp)
    {
        $this->id = $id;
        $this->authorId = $authorId;
        $this->content = $content;
        $this->done = $done;
        $this->timestamp = $timestamp;
    }

    public function jsonSerialize()
    {
        return [
        'id' => $this->id,
        'authorId' => $this->authorId,
        'content' => $this->content,
        'done' => $this->done,
        'timestamp' => $this->timestamp
        ];
    }
}

File: android/app/src/main/kotlin/pl/kwasow/data/types/Wish.kt
package pl.kwasow.data.types

import kotlinx.serialization.Serializable

@Serializable
data class Wish(
    val id: Int,
    val authorId: Int,
    val content: String,
    val done: Boolean,
    val timestamp: Long,
) {
    // ====== Public methods
    fun update(
        newContent: String = content,
        newDone: Boolean = done,
    ): Wish = Wish(id, authorId, newContent, newDone, timestamp)
}
